tlatools/org.lamport.tlatools/src/tlc2/value/impl/LazyValue.java

```diff
@@ -121,6 +117,10 @@ public LazyValue(SemanticNode expr, Context con, final boolean cachable, final C
     }
   }
 
+  public CostModel getCostModel() {
```

Comment:
Do you generally think it is necessary to define getter methods for public final fields?

Reply:
Definitely not. Good style dictates this should be a getter; performance dictates this should be a public field; we have both because `Value` has both and I don't want to fix all the usages just yet.

Reply:
I've added a TODO for this.